src/Microsoft.DotNet.Build.Tasks/PackageFiles/ApiCompat.targets

```diff
@@ -0,0 +1,134 @@
+  <PropertyGroup Condition="'$(AssemblyName)' != ''">
+    <AssemblyVersion Condition="'$(AssemblyVersion)'==''">4.0.0.0</AssemblyVersion>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <ApiCompatBaseline>$(MSBuildProjectDirectory)\ApiCompatBaseline.txt</ApiCompatBaseline>
+    <ImplementationOutputPath>$(BaseOutputPath)$(OSGroup).$(Platform).$(ConfigurationGroup)\$(AssemblyName)\$(targetGroup)</ImplementationOutputPath>
+    <ImplementationProjectJson Condition="'$(ImplementationProjectJson)'=='' and Exists('$(MSBuildProjectDirectory)/project.json')">$(MSBuildProjectDirectory)/project.json</ImplementationProjectJson>
+    <ImplementationProjectLockJson Condition="Exists('$(ImplementationProjectJson)') and '$(ImplementationProjectLockJson)'==''">$(MSBuildProjectDirectory)/project.lock.json</ImplementationProjectLockJson>
+    <ReferenceAssembly>$(BaseOutputPath)ref\$(AssemblyName)\$(AssemblyVersion)\$(AssemblyName).dll</ReferenceAssembly>
+    <ReferenceProjectDirectory>$(MSBuildProjectDirectory)\..\ref</ReferenceProjectDirectory>
+    <ReferenceProject>$(ReferenceProjectDirectory)\$(MSBuildProjectFile)</ReferenceProject>
+    <ReferenceOutputPath>$(ReferenceAssemblyOutputPath)$(AssemblyName)/$(AssemblyVersion)</ReferenceOutputPath>
+    <ReferenceProjectJson Condition="'$(ReferenceProjectJson)'=='' and Exists('$(ReferenceProjectDirectory)/project.json')">$(ReferenceProjectDirectory)/project.json</ReferenceProjectJson>
+    <ReferenceProjectLockJson Condition="Exists('$(ReferenceProjectJson)') and '$(ReferenceProjectLockJson)'==''">$(ReferenceProjectDirectory)/project.lock.json</ReferenceProjectLockJson>
+  </PropertyGroup>
+  
+  <!-- ApiCompat for Implementation Assemblies  -->
+  <Target Name="ValidateApiCompatForSrc"
+          AfterTargets="Build;CopyFilesToOutputDirectory"
+          DependsOnTargets="GetDependencyDirectories;GetContractDependencies" 
+          Condition="Exists('$(ReferenceProject)')" >
+    
+    <PropertyGroup Condition="$(MSBuildProjectDirectory.EndsWith('src')) AND Exists('$(ReferenceProjectLockJson)')">
+      <_ApiCompatCmd>"$(ToolsDir)CoreRun.exe" "$(ToolsDir)ApiCompat.exe" "$(ReferenceAssembly)" </_ApiCompatCmd>
+      <_ApiCompatCmd>$(_ApiCompatCmd) -contractDepends:"@(_ContractDependencies);"</_ApiCompatCmd>
+      <_ApiCompatCmd>$(_ApiCompatCmd) -implDirs:"$(ImplementationOutputPath);@(_DependencyDirectories);"</_ApiCompatCmd>
+      <_ApiCompatCmd Condition="Exists('$(ApiCompatBaseline)')">$(_ApiCompatCmd) -baseline:"$(ApiCompatBaseline)"</_ApiCompatCmd>
+    </PropertyGroup>
+    <Exec Condition="$(MSBuildProjectDirectory.EndsWith('src')) AND Exists('$(ReferenceProjectLockJson)')" 
+          Command="$(_ApiCompatCmd) "
+          CustomErrorRegularExpression="^[a-zA-Z]+ :" />
+    <Warning Condition="!Exists('$(ReferenceProjectLockJson)')" Text="The project.lock.json was not found for : $(ReferenceProject)" />      
```

Comment:
Is this working in the right location? It seems like the call to ApiCompat will be completely invalid if this doesn't exist.